### DISCUSSION TITLE
Item should have attribute for supported payment modes

### DISCUSSION THREAD

**Discussion by BLR-0118** [beckn/protocol-specifications#282] [Core Specification]
Buyers often prefer to select items, from a catalog, based on the payment criteria (e.g. cash, digital). Currently, when catalog displays an item, it is not clear what the payment criteria is for that item. Item should support an attribute for supported_payment_modes with enums of "Cash", "Digital".

**Comment by BLR-0118** [comment]
Pramod, tagging you since Ravi suggested you have thoughts on this
@pramodkvarma

**Comment by pramodkvarma** [comment]
I thought we did have attributes that can carry promotional tags. Could we use that instead of fully structured payment object? Also, remember to make catalogue caching easier and have all dynamic info come through during select call. But if we can't leverage existing tags we can explore additional tags. @core-wg-admin

**Comment by ravi-prakash-v** [comment]
@pramodkvarma @BLR-0118 FYR, the current `Provider` schema in `0.9.4-draft` is shown [here](https://github.com/beckn/protocol-specifications/blob/e9a72d5d037c06e862ef69e0b6ba413687c429b7/core/v0/api/core.yaml#L2252)

As you can see it has a property called `payments` that is an array of objects of type `Payment`.

Each of these `Payment` objects contain a unique payment policy of a **Provider**.

I believe it would be more useful to tag each item with a reference to a particular Payment _policy_ rather than a single attribute of payment i.e _payment mode_. That would make it more generic and would allow a larger set of use cases to be covered.

However to enabled this, there are two changes to be made here.

Firstly, the `Payment` object must have an `id` property so that it can be referenced at other places. 

Secondly, the `Item` schema must have a reference to a `Payment` object inside the `payments` array using a property called `payment_id`.

Similarly, the `Provider` schema must also have a `payment_id` property that allows a default `Payment` policy to be defined all across the provider's catalog overriden only by the `payment_id` at the `Item` level.

**Comment by pramodkvarma** [comment]
Would like to see a sample json before and after (based on above proposal) to make it easier for me to fully understand. Or if there is wiki/doc page on it point me to that. Thanks.
